homeassistant/config_entries.py

```diff
@@ -1314,6 +1314,22 @@ def _data_to_save(self) -> dict[str, list[dict[str, Any]]]:
         """Return data to save."""
         return {"entries": [entry.as_dict() for entry in self._entries.values()]}
 
+    async def async_wait_entry(self, entry: ConfigEntry) -> bool:
+        """Wait for an entry's component to load and return if the entry is loaded.
+
+        This is primarily intended for existing config entries which are loaded at
+        startup, awaiting this function will block until the component and all its
+        config entries are loaded.
+        Config entries which are created after Home Assistant is started can't be waited
+        for, the function will just return if the config entry is loaded or not.
+        """
```

Comment:
While this all makes a whole lot of sense to me, I do think the method name is misleading. It is not waiting for an entry to load; it is waiting for a component to load.

Reply:
Good point, renamed to `async_wait_component` 👍